File: aulas/programacao orientada a objetos/exemplos/python/calculo_tarifa.py

class ProdutoImportado:

    def __init__(self, ncm, descricao, valor_cif, aliquota):
        self.ncm = ncm
        self.descricao = descricao
        self.valor_cif = valor_cif
        self.aliquota = aliquota

    def tarifa_importacao(self):
        return self.valor_cif * self.aliquota

    def valor_final(self):
        return self.valor_cif + self.tarifa_importacao()


produtos = [
    ProdutoImportado("8471.30.12","Notebook",1000,0.14),
    ProdutoImportado("8517.12.31","Smartphone",800,0.16),
    ProdutoImportado("8528.72.00","Televisor",1200,0.20)
]

for p in produtos:
    print(p.descricao, p.valor_final())

File: aulas/programacao orientada a objetos/exemplos/r/calculo_tarifa.R

library(R6)

ProdutoImportado <- R6Class(
 "ProdutoImportado",
 public=list(
  ncm=NULL,
  descricao=NULL,
  valor_cif=NULL,
  aliquota=NULL,

  initialize=function(ncm,descricao,valor_cif,aliquota){
   self$ncm <- ncm
   self$descricao <- descricao
   self$valor_cif <- valor_cif
   self$aliquota <- aliquota
  },

  tarifa_importacao=function(){
   self$valor_cif*self$aliquota
  },

  valor_final=function(){
   self$valor_cif+self$tarifa_importacao()
  }
 )
)

notebook <- ProdutoImportado$new("8471.30.12","Notebook",1000,0.14)
notebook$valor_final()
